prov/verbs/src/verbs_info.c

```diff
@@ -1208,9 +1208,21 @@ static int fi_ibv_get_matching_info(uint32_t version,
 	*info = tail = NULL;
 
 	for ( ; check_info; check_info = check_info->next) {
+		VERBS_DBG(FI_LOG_FABRIC, "Checking domain: %s\n",
+			  check_info->domain_attr->name);
+
 		if (hints) {
-			VERBS_DBG(FI_LOG_FABRIC, "Checking domain: %s\n",
-				  check_info->domain_attr->name);
+			if ((check_info->ep_attr->protocol ==
+			     FI_PROTO_RDMA_CM_IB_XRC) &&
+			    (!hints->ep_attr ||
+			     (hints->ep_attr->rx_ctx_cnt != FI_SHARED_CONTEXT))) {
```

Comment:
With this change, I can no longer get RxM to get/select an XRC info. If FI_OFI_RXM_USE_SRX is set, is there a way to have RxM setup the core-hints to pick up the XRC fi_info?

Reply:
Can you try setting FI_OFI_RXM_USE_SRX=1?

Reply:
Yes I have been running with FI_OFI_RXM_USE_SRX set. If I do a simple "fi_info" with no parameters, "verbs;ofi_rxm" provider is only displayed for the RC endpoint: e.g.:

[swelch@roce-j libfabric]$ fi_info
provider: verbs;ofi_rxm
    fabric: IB-0xfe80000000000000
    domain: mlx5_0
    version: 1.0
    type: FI_EP_RDM
    protocol: FI_PROTO_RXM
provider: verbs
    fabric: IB-0xfe80000000000000
    domain: mlx5_0-xrc
    version: 1.0
    type: FI_EP_MSG
    protocol: FI_PROTO_RDMA_CM_IB_XRC
provider: verbs
    fabric: IB-0xfe80000000000000
    domain: mlx5_0
    version: 1.0
    type: FI_EP_MSG
    protocol: FI_PROTO_RDMA_CM_IB_RC
provider: verbs
    fabric: IB-0xfe80000000000000
    domain: mlx5_0-dgram
    version: 1.0
    type: FI_EP_DGRAM
    protocol: FI_PROTO_IB_UD
provider: verbs;ofi_rstream
    fabric: IB-0xfe80000000000000
    domain: mlx5_0
    version: 1.6
    type: FI_EP_SOCK_STREAM
    protocol: FI_PROTO_RSTREAM
[swelch@roce-j libfabric]$

When using RxM is never sees the XRC info, always using the RC info.

Reply:
@shefty - I see the error, in rxm_init.c:111 the check should be !fi_param_get_bool() instead of fi_param_get_bool(). It works great  if I make that change.